src/Microsoft.ML.Featurizers/TimeSeriesImputer.cs

```diff
@@ -559,6 +570,8 @@ internal static TypedColumn CreateTypedColumn(DataViewSchema.Column column, stri
                     return new NumericTypedColumn<double>(column, optionalColumns.Contains(column.Name));
                 else if (type == typeof(ReadOnlyMemory<char>).ToString())
                     return new StringTypedColumn(column, optionalColumns.Contains(column.Name));
+                else if (type == typeof(DateTime).ToString())
```

Comment:
This `if` clause seems like it should be re-written to not call `ToString()` and compare strings.

```C#
var type = column.Type.RawType;
if (type == typeof(sbyte))
{
...
}
else if (type == typeof(short))
{
...
```

That is a much more performant way of doing the same checks.

Reply:
This check happens only once for each column, so its not a big performance hit. It also makes saving and loading much easier due to namespace issues. I had an issue in the past with this when I was comparing types directly. If you think its a big enough deal I can refactor it, but I think the amount of perf we will gain will be negligible since its only once per column.